modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,70 @@
+import { RealKeys } from '@ephox/agar';
+import { before, context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { PlatformDetection } from '@ephox/sand';
+
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('webdriver.tinymce.core.keyboard.PageUpDownKeyTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const platform = PlatformDetection.detect();
+  const supportsPageUpDown = !(platform.os.isOSX() || platform.os.isWindows() && platform.browser.isFirefox());
+
+  // It necessary to skip tests for Mac and windows Firefox they don't move the selection when pressing page up/down
+  context('Page Up', () => {
+    before(function () {
```

Comment:
This could be moved outside the `context` blocks and that way you wouldn't need to duplicate it. eg:
```
  // It's necessary to skip tests for Mac and Windows Firefox as they don't move the selection when pressing page up/down
  before(function () {
    if (!supportsPageUpDown) {
      this.skip();
    }
  });

  context('Page Up', () => {
    ...
  });

  context('Page Down', () => {
    ...
  });
```

Note: I also fixed some minor grammar issues in your comment in the above.

Reply:
updated